src/control/lib/control/system.go

```diff
@@ -744,3 +744,82 @@ func PingRanks(ctx context.Context, rpcClient UnaryInvoker, req *RanksReq) (*Ran
 
 	return invokeRPCFanout(ctx, rpcClient, req)
 }
+
+// SystemCleanupReq contains the inputs for the system cleanup request.
+type SystemCleanupReq struct {
+	unaryRequest
+	msRequest
+	sysRequest
+	Machine string `json:"machine"`
+}
+
+type CleanupCount struct {
+	Id    string `json:"Id"`    // Unique identifier
+	Count uint32 `json:"count"` // Number of pools reclaimed
+}
+
+// SystemCleanupResp contains the request response.
+type SystemCleanupResp struct {
+	Status int32           `json:"status"`
+	Pools  []*CleanupCount `json:"pools"`
+}
+
+// Validate returns error if response contents are unexpected, string of
+// warnings if pool queries have failed or nil values if contents are expected.
+func (scr *SystemCleanupResp) Validate() (string, error) {
+	out := new(strings.Builder)
+
+	for i, p := range scr.Pools {
+		if p.Id == "" {
+			return "", errors.Errorf("pool with index %d has no Id", i)
+		}
+	}
+
+	return out.String(), nil
+}
+
+// Errors returns a single error combining all error messages associated with a
+// system cleanup response.
+func (scr *SystemCleanupResp) Errors() error {
+	warn, err := scr.Validate()
+	if err != nil {
+		return err
+	}
+	if warn != "" {
+		return errors.New(warn)
+	}
+	return nil
+}
+
+// SystemCleanup requests resources associated with a machine name be cleanedup.
+//
+// Handles MS requests sent from management client app e.g. 'dmg' and calls into
+// mgmt_system.go method of the same name. The triggered method uses the control
+// API to fanout to (selection or all) gRPC servers listening as part of the
+// DAOS system and retrieve results from the selected ranks hosted there.
```

Comment:
I'm happy to remove it. I was just giving a similar level of detail compared to what I already saw in the file.

Reply:
Understood. My instinct is to remove it rather than try to update to describe communications with the MS. If we want to provide those details to someone who isn't reading the code, it should probably be in our high-level docs.

Reply:
Ok I'll remove it.